/**
Copyright 2018 BarD Software s.r.o

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto3";

message FileVersionInfo {
    int64 generation = 1;
    int64 timestamp = 2;
    string fileId = 3;
    string userName = 4;
    string versionName = 5;
}

message FileVersion {
    repeated Patch patches = 1;
    bytes content = 2;
    int64 timestamp = 3;
    repeated FileVersionInfo historyWindow = 4;
    string fileId = 5;
}

message PatchList {
    repeated Patch patches = 1;
}

message Patch {
    string userId = 1;
    string text = 2;
    int64 timestamp = 3;
    string actualHash = 4;
    string userName = 6;
}
